### ISSUE TITLE
`kubeadm/kubernetes-xenial,now 1.10.6-00 amd64` depends on crictl, renders join unusable

### ISSUE THREAD

**Issue by alvaroaleman** [kubernetes/kubeadm#1038]
### BUG REPORT

This now means that `kubeadm join` is broken by default in the `1.10` apt package, the only workaround is running the `kubeadm join` with `--ignore-preflight-errors=CRI`

## Versions

**kubeadm version** (use `kubeadm version`):
`kubeadm version: &version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.6", GitCommit:"a21fdbd78dde8f5447f5f6c331f7eb6f80bd684e", GitTreeState:"clean", BuildDate:"2018-07-26T10:04:08Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"`

**Environment**:
- **Kubernetes version** (use `kubectl version`):
- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
```
NAME="Ubuntu"
VERSION="16.04.5 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
- **Kernel** (e.g. `uname -a`):
- **Others**:

## What happened?

Installed `kubeadm/kubernetes-xenial,now 1.10.6`, tried to run `kubeadm join`, this failed with 
```
[ERROR CRI]: unable to check if the container runtime at "/var/run/dockershim.sock" is running: fork/exec /usr/local/bin/crictl -r /var/run/dockershim.sock info: no such file or directory
```

The original issue for this is #814 which got fixed but not backported to 1.10.
The problem is that the 1.10.6 version of the kubeadm apt pacakge added a dependency on `cri-tools` with includes `crictl`:

```
apt-cache depends kubeadm=1.10.6-00
kubeadm
  Depends: kubelet
  Depends: kubectl
  Depends: kubernetes-cni
  Depends: cri-tool
```

Versus the 1.10.5 version which works fine:

```
apt-cache depends kubeadm=1.10.5-00
kubeadm
  Depends: kubelet
  Depends: kubectl
```

## What you expected to happen?

The `kubeadm` package not to pull in the `cri-tools` package OR having a bugfix for #814 in the `1.10` version

## How to reproduce it (as minimally and precisely as possible)?

See above

**Comment by neolit123** [comment]
crictl is required if you are not using docker.

but we can cherry pick this for 1.10.x to solve the issue for docker users.
https://github.com/kubernetes/kubernetes/pull/66282

do you think this is a good idea?
@kad, @bart0sh, @chuckha

**Comment by neolit123** [comment]
@alvaroaleman 

```
apt-cache depends kubeadm=1.10.6-00
kubeadm
  Depends: kubelet
  Depends: kubectl
  Depends: kubernetes-cni
  Depends: cri-tool
```

is the package name `cri-tool` really here?
should be `cri-tools`.

**Comment by alvaroaleman** [comment]
@neolit123 I guess that I missed the `s` when copying. I know `crictl` is required for non-docker users, however current state is broken by default for docker users in the 1.10 branch

**Comment by chuckha** [comment]
I think the root cause is the release repo, but I need more insight to how previous releases get released  to guarantee it. I need to know how 1.10 gets built when 1.11 is the stable version.

[This line](https://github.com/kubernetes/release/blob/master/debian/xenial/kubeadm/debian/control#L13) isn't bounded by version. I think release is always run from the master branch and a version of k8s is passed in to build.go. This means 1.10.x does not have the code needed to depend on cri-tools, but the debs/rpms are built depending on cri-tools.

I think in this case, a cherry pick is what we want but only because this won't be an issue in 2 releases.

**Comment by timothysc** [comment]
This should be fixed for 1.11+

**Comment by praseodym** [comment]
Does this mean Kubernetes ≥1.10.6, <1.11 is not supported by kubeadm any more?
